### PR TITLE
Update capi to match HELICS3

### PR DESCRIPTION
**Pull request by kdheepak**

### CODE REVIEW

File: helics/capi.py [hunk 1]
```diff
@@ -3119,77 +3239,83 @@ def helicsEndpointGetDefaultDestination(endpoint: HelicsEndpoint) -> str:
...
             """Raw data must be of type `bytes`. Got {t} instead. Try converting it to bytes (e.g. `"hello world".encode()`""".format(t=type(data))
         )
     inputDataLength = len(data)
-    f(endpoint.handle, cstring(dest), data, inputDataLength, err)
+    f(endpoint.handle, data, inputDataLength, err)
     if err.error_code != 0:
         raise HelicsException("[" + str(err.error_code) + "] " + ffi.string(err.message).decode())
 
 
-def helicsEndpointSendEventRaw(
```
Comment (kdheepak) [inline helics/capi.py:3144]: @phlptp is this function removed?

File: helics/_build.py [hunk 2]
```diff
@@ -42,13 +42,13 @@
 # ffi.set_source(
 #     "_py_helics",
 #     f"""#include "{CHELICS}" """,
-#     libraries=["helicsSharedLib"],
+#     libraries=["libhelics"],
```
Comment (phlptp) [inline helics/_build.py:45]: is the lib supposed to be on this?

Comment (kdheepak) [reply helics/_build.py:45]: Yes!

Comment (phlptp) [reply helics/_build.py:45]: why wasn't it on previously?

Comment (kdheepak) [reply helics/_build.py:45]: I think I may have made the change locally but didn't push.

**Comment by phlptp** [comment]
Is there something in particular you want me to look at otherwise I am not sure how to review this?

**Comment by kdheepak** [comment]
Not at the moment, when I make the final PR I'll comment on the specific changes that I want you to see. I've already discussed these changes with you on conference calls, so we should be good.

**Pull request closed without merging**